### ISSUE TITLE
Add confirmation prompt like in apt, pacman, etc.

### ISSUE THREAD

**Issue by robinhood2014** [microsoft/winget-cli#1270]
# Description of the new feature/enhancement

# Proposed technical implementation details (optional)

**Comment by rbarbrow** [comment]
I would counter this argument. In windows no one is going to Accidently type "winget uninstall foo"  or "winget install Foo.Foo". I think a prompt would just add more layers of complexity when trying to script

**Comment by KaranKad** [comment]
I agree with @rbarbrow. Confirmation prompt should only pop up if application has some dependency and want's to download it. https://github.com/microsoft/winget-cli/issues/163#issuecomment-633901489

**Comment by robinhood2014** [comment]
...which is a pretty smart idea. Apt proceeds with the installation without confirmation if the user-specified package is the only package that it has to install; however, it asks for confirmation in all other install/upgrade cases, and in all removal cases.

**Comment by KaranKad** [comment]
Still, this feature could be added as a optional opt-in feature in `winget settings` for those who want it.

**Comment by robinhood2014** [comment]
That's an excellent idea. Maybe users who are coming to Windows from Linux will be more comfortable with WinGet operating that way.On Jul 12, 2021 1:07 PM, Karan09 ***@***.***> wrote:
Still, this feature could be added as a optional opt-in feature in winget settings for those who want it.

**Comment by ksast** [comment]
I recently encountered a situation where this would have been very useful. I am currently using winget to schedule automatic package upgrades. Typically, I utilize winget upgrade with the --silent flag, allowing packages to be upgraded in the background without disrupting the user. However, there was an instance where the package JGraph.Draw was updated while the application was running. The user received a brief popup prompting them to save their work, but the upgrade continued before they had a chance to actually save. Having a user confirmation for this particular application would have been beneficial in preventing any potential data loss.
